tests/parsers/test_pw.py

```diff
@@ -496,6 +498,64 @@ def test_pw_failed_interrupted_xml(
     data_regression.check(results['output_parameters'].get_dict())
 
 
+@pytest.mark.parametrize(
+    'test_case, expected_exit_code', (
+        ('default', None),
+        ('failed_interrupted', PwCalculation.exit_codes.ERROR_SCHEDULER_OUT_OF_WALLTIME),
+    )
+)
+def test_pw_failed_interrupted_scheduler(
+    fixture_localhost, generate_calc_job_node, generate_parser, generate_inputs, test_case, expected_exit_code
+):
+    """Test that an exit code set by the scheduler is not overridden unless a more specific error is parsed.
+
+    The test is run twice, once for the ``default`` test case and once for ``failed_interrupted``, which correspond to a
+    successful run and a run that got interrupted (usually due to scheduler killing the job). Before calling the parser
+    the ``ERROR_SCHEDULER_OUT_OF_WALLTIME`` is set on the node. In the case of the ``default`` test case, this should be
+    ignored and the parser should return ``ExitCode(0)``. For the interrupted case, the exit code of the scheduler
+    should not be overridden so the parser should return ``None``.
```

Comment:
Doesn't the parser return `PwCalculation.exit_codes.ERROR_SCHEDULER_OUT_OF_WALLTIME` in this case?